client/src/routes/challs.js

```diff
@@ -254,13 +270,13 @@ const Challenges = ({ classes }) => {
           <div class='frame__body'>
             <div class='frame__title title'>Tags</div>
             {
-              Object.keys(tags).sort((a, b) => a.localeCompare(b)).map((metatag) => {
+              Array.from(tags.keys()).sort((a, b) => a.localeCompare(b)).map((metatag) => {
                 return (<div><h5 class='frame__title title'>{metatag}</h5>
-                  {Object.entries(tags[metatag]).sort((a, b) => a[0].localeCompare(b[0])).map(([tag, checked]) => {
+                  {Array.from(tags.get(metatag).entries()).sort((a, b) => a[0].localeCompare(b[0])).map(([tag, checked]) => {
                     return (
-                      <div key={tag} class='form-ext-control form-ext-checkbox'>
-                        <input id={`tag-${tag}`} data-tag={tag} data-metatag={metatag} class='form-ext-input' type='checkbox' checked={checked} onChange={handleTagsCheckedChange} />
-                        <label for={`tag-${tag}`} class='form-ext-label'>{tag}</label>
+                      <div key={`tag-${metatag}-${tag}`} class='form-ext-control form-ext-checkbox'>
```

Comment:
Technically not unique (e.g. metatag `a-b` tag `c` and metatag `a` tag `b-c` both become `tag-a-b-c`) but it's a bit annoying to fix. One way to dedupe is by inserting the metatag's length e.g. `` `tag-${metatag.length}-${metatag}-${tag}` ``